py/desiutil/dust.py

```diff
@@ -30,23 +31,44 @@ def extinction_total_to_selective_ratio(band, photsys) :
     Returns:
         scalar, total extinction A(band) = -2.5*log10(transmission(band))
     """
+    if match_legacy_surveys :
+        # Based on the fit from the columns MW_TRANSMISSION_X and EBV
+        # for the DR8 target catalogs and propagated in fibermaps
+        # R_X = -2.5*log10(MW_TRANSMISSION_X) / EBV
+        # It is the same value for the N and S surveys in DR8 and DR9 catalogs.
+        R={"G_N":3.2140,
+           "R_N":2.1650,
+           "Z_N":1.2110,
+           "G_S":3.2140,
+           "R_S":2.1650,
+           "Z_S":1.2110,
+        }
+    else :
+        # From https://desi.lbl.gov/trac/wiki/ImagingStandardBandpass
+        # DECam u  3881.6   3.994
+        # DECam g  4830.8   3.212
+        # DECam r  6409.0   2.164
+        # DECam i  7787.5   1.591
+        # DECam z  9142.7   1.211
+        # DECam Y  9854.5   1.063
+        # BASS g  4772.1   3.258
+        # BASS r  6383.6   2.176
+        # MzLS z  9185.1   1.199
+        # Consistent with the synthetic magnitudes and function dust_transmission
+
+        R={"G_N":3.258,
+           "R_N":2.176,
+           "Z_N":1.199,
+           "G_S":3.212,
+           "R_S":2.164,
+           "Z_S":1.211,
+        }
 
-    # Based on the fit from the columns MW_TRANSMISSION_X and EBV
-    # for the DR8 target catalogs and propagated in fibermaps
-    # R_X = -2.5*log10(MW_TRANSMISSION_X) / EBV
-    # excerpt from https://www.legacysurvey.org/dr8/catalogs/#galactic-extinction-coefficients :  Eddie Schlafly has computed the extinction coefficients for the DECam filters through airmass=1.3, computed for a 7000K source spectrum as was done in the Appendix of Schlafly & Finkbeiner (2011). These coefficients are A / E(B-V) = 3.995, 3.214, 2.165, 1.592, 1.211, 1.064 (note that these are slightly different from the coefficients in Schlafly & Finkbeiner 2011). The coefficients are multiplied by the SFD98 E(B-V) values at the coordinates of each object to derive the g, r and z mw_transmission values in the Legacy Surveys catalogs. The coefficients at different airmasses only change by a small amount, with the largest effect in g -band where the coefficient would be 3.219 at airmass=1 and 3.202 at airmass=2. We calculate Galactic extinction for BASS and MzLS as if they are on the DECam filter system.
-
-    R={"G_N":3.2140,
-       "R_N":2.1650,
-       "Z_N":1.2110,
-       "G_S":3.2829,
-       "R_S":2.1999,
-       "Z_S":1.2150}
     assert(band.upper() in ["G","R","Z"])
     assert(photsys.upper() in ["N","S"])
```

Comment:
Let's also support PHOTSYS='G' (Gaia) since that appears in the targeting catalogs as well, with allowable band values of "G", "BP", and "RP".  This would move `desispec.scripts.stdstars.unextinct_gaia_mags` logic over here instead of having Gaia treated as a special case on the desispec side (which I think it still is there in desihub/desispec#1246).

For consideration: older targets/fiberassign files also have PHOTSYS='' meaning Gaia; we should track down whether that is recent enough for us to care (Dec 2020 or later), but I haven't done that yet.

Reply:
@segasai could you help with moving the Gaia extinction code out of desispec and into this PR branch so that `desiutil.dust` is our one-stop-shop for extinction corrections?

Reply:
I'll see what I can do (on a timescale of few days)